stl/inc/span

```diff
@@ -0,0 +1,665 @@
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rend() const noexcept {
+        return const_reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crbegin() const noexcept {
+        return rbegin();
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crend() const noexcept {
+        return rend();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_begin() noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_begin() const noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_end() noexcept {
+        return _Mydata + size();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_end() const noexcept {
+        return _Mydata + size();
+    }
+
+private:
+    pointer _Mydata{nullptr};
+};
+
+// [span.tuple] Tuple interface
+template <class _Ty, size_t _Extent>
+struct tuple_size<span<_Ty, _Extent>> : integral_constant<size_t, _Extent> {
+    static_assert(dynamic_extent != _Extent, "std::span<T, dynamic_extent> is not tuple-like. ");
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+struct tuple_element<_Index, span<_Ty, _Extent>> {
+    static_assert(dynamic_extent != _Extent, "std::span<T, dynamic_extent> is not tuple-like. ");
+    static_assert(_Index < _Extent, "Index out of bounds for a std::span of this extent. ");
+    using type = _Ty;
+};
+
+template <size_t _Index, class _Ty, size_t _Extent>
+_NODISCARD constexpr _Ty& get(span<_Ty, _Extent> _Span) noexcept {
+    static_assert(dynamic_extent != _Extent, "std::get<> not supported for std::span<T, dynamic_extent>");
+    static_assert(_Index < _Extent, "Index out of bounds for a std::span of this extent. ");
+    return _Span[_Index];
+}
+
+#if _HAS_STD_BYTE
+// [span.objectrep] Views of object representation
+// FUNCTION TEMPLATE as_bytes
+template <class _Ty, size_t _Extent, enable_if_t<_Extent <= dynamic_extent / sizeof(_Ty), int> = 0>
+_NODISCARD auto as_bytes(span<_Ty, _Extent> _Sp) noexcept -> span<const byte, sizeof(_Ty) * _Extent> {
+    return {reinterpret_cast<const byte*>(_Sp.data()), _Sp.size_bytes()};
+}
+
+template <class _Ty>
+_NODISCARD auto as_bytes(span<_Ty, dynamic_extent> _Sp) noexcept -> span<const byte, dynamic_extent> {
+    return {reinterpret_cast<const byte*>(_Sp.data()), _Sp.size_bytes()};
+}
+
+// FUNCTION TEMPLATE as_writable_bytes
+template <class _Ty, size_t _Extent, enable_if_t<!is_const_v<_Ty> && _Extent <= dynamic_extent / sizeof(_Ty), int> = 0>
+_NODISCARD auto as_writable_bytes(span<_Ty, _Extent> _Sp) noexcept -> span<byte, sizeof(_Ty) * _Extent> {
+    return {reinterpret_cast<byte*>(_Sp.data()), _Sp.size_bytes()};
+}
+
+// FUNCTION TEMPLATE as_writable_bytes
+template <class _Ty, enable_if_t<!is_const_v<_Ty>, int> = 0>
+_NODISCARD auto as_writable_bytes(span<_Ty, dynamic_extent> _Sp) noexcept -> span<byte, dynamic_extent> {
+    return {reinterpret_cast<byte*>(_Sp.data()), _Sp.size_bytes()};
+}

```

Comment:
Same comment about overloading and overflow.

Reply:
Done